calico-cloud/image-assurance/scanners/registry-scanner.mdx

```diff
@@ -0,0 +1,150 @@
+- `AZURE_CLIENT_ID`
+- `AZURE_CLIENT_SECRET` or `AZURE_FEDERATED_TOKEN`
+- `AZURE_TENANT_ID`
+
+**AWS credentials required**
+- `AWS_ACCESS_KEY_ID`
+- `AWS_SECRET_ACCESS_KEY`
+- `AWS_REGION`
+
+The registry scanner currently scans one registry at a time. If you specify the Azure or AWS credentials, the Docker username and password will not be used.
+
+#### Required inputs
+
+The registry scanner requires:
+
+- `REGISTRY` - the registry you want to scan. (gcr.io)
+- `IMAGE_ASSURANCE_API_URL` - Get the URL in the Manager UI.
+- `IMAGE_ASSURANCE_API_TOKEN` - Get the token in the Manager UI.
+- Either the `DOCKER_USERNAME` and `DOCKER_PASSWORD`, Azure credentials, or AWS credentials.
+
+#### Optional inputs
+
+`REGISTRY_FILTER` - limits scanning time when you have thousands of repositories and images. Supports a comma-separated list.
+
+Example: for a gcr registry that contains:
+```
+gcr.io/prod-env/api
+gcr.io/staging-env/api
+gcr.io/dev/api
+```
+To filter out images in the dev "sub" registry, use:  `-e REGISTRY_FILTER=prod-env,staging-env`.
+
+### Scan images and send results to {{prodname}}
+
+Example run for a gcr registry with Docker Username and Password:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e DOCKER_USERNAME=<username> -e DOCKER_PASSWORD=<password>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run for an acr registry with Azure credentials:
+
+```
+docker run -e REGISTRY=your-org.azurecr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e AZURE_CLIENT_ID=<azure-client-id> -e AZURE_CLIENT_SECRET=<azure-client-secret> -e AZURE_TENANT_ID=<azure-tenant-id>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run for an ecr registry with AWS credentials:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -e AWS_ACCESS_KEY_ID=<aws-access-key> -e AWS_SECRET_ACCESS_KEY=<aws-secret-access-key> -e AWS_REGION=<aws-region>
+ quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+Example run when mounting the dockerfile:
+
+```
+docker run -e REGISTRY=gcr.io -e IMAGE_ASSURANCE_API_URL=https://<clustId>-management.dev.calicocloud.io/bast
+ -e IMAGE_ASSURANCE_API_TOKEN=$TOKEN -v ~/.docker/config.json/:/.docker/config.json quay.io/tigera/image-assurance-registry-scanner:v1.10.4
+```
+
+### Troubleshoot
+
+- For issues authenticating with the registry, verify that the registry credentials are correct.
+- For issues uploading the results to the {{prodname}}, check that the API token and URL are correct.
+- Because the scanner is not running continuously, it may take some time for results to display in the Image Assurance dashboard, especially if you have many images or large images.
```

Comment:
Can you clarify here what "scanner" you're referring to? This doesn't sound like you're describing the registry scanner. 

If you are, then I find the "not running continuously" curious. Technically once it starts executing, it is running continuously, until it was scanned all of the images. 

So from a user perspective, I'm not quite sure what this is trying to say.

Reply:
This was in reference to the scanner worker... I'm wondering how we should word this. 

Perhaps: `Images scanned using the registry scanner may take up to 5 minutes to appear on the Manager UI`?

Reply:
It is indeed a bit tricky to word this with enough information without getting into too much detail about what scanner worker is and how it relates to the registry scanner. 

Perhaps we can just say something like "Image scan results uploaded to Calico Cloud through the registry scanner   require additional processing before appearing in the UI. This may result in a time delay before CVE results appear for those images in the UI."

Reply:
Fixed.